### ISSUE TITLE
future development

### ISSUE THREAD

**Issue by weiliu620** [openai/rllab#157]
This is rather vague question: what is the future plan of rllab? 

I'm new to RL. While learning and implementing some algorithms, I think rllab is a great platform that I can add my own code. But, I'm not sure I should invest time reading rllab's code, unless I know there will be on-going development and maintenance on rllab in the long term. More specifically, 

I see Tensorflow implementation in sandbox. Will TF replace Theano in the future?

I also see a few environments implemented, which is not exactly same with Gym's. For example, the environment parameters and reward definition of cart pole environment is different from Gym. Which one shall I focus on? It seems a redundant effort to add environments in rllab while Gym already has it? 

There is no official release yet, at least not from Github. Does it mean the code is going through big change. Do you plan to release a stable version any time soon? That will give me confidence that I should spend time on it. 

Thanks,

**Comment by ViktorM** [comment]
Good questions, I can add one more:

What is a relationship with OpenAI Baselines? They claimed that **"OpenAI Baselines is a set of high-quality implementations of reinforcement learning algorithms."** So what is a difference, which library will become the main RL library in future? Which one will be supported and continue development?

**Comment by dementrock** [comment]
Good question indeed. Currently there is no plan to actively develop rllab further, although a few people at OpenAI and Berkeley (including myself) are still using it for ongoing research projects, and I will try my best to answer questions here. I expect OpenAI Baselines should be more well-maintained and more production quality.

For environments, you should aim for Gym compatibility, since rllab also supports Gym. The existing environments in rllab are kept for convenience and parameters are not updated to be consistent with what we did in the paper.

**Comment by ViktorM** [comment]
@dementrock , thanks for clarification. I can say rllab is of very good quality itself. The only difference so far I noticed - Baselines started working out of the box on Windows 10 just after "pip install baselines". ) It's quite important for us, as our research is required multiplatform support, Windows 10 and Linux.

Will Baselines be based only on TF or you plan to add support for some other DL libraries, PyTorch for example? Can you tell, at least approximately when the first support of continuous control algorithms - TRPO, DDPG, Q-prop will added to Baselines?

**Comment by dementrock** [comment]
@ViktorM None of the rllab users at OpenAI / Berkeley use windows unfortunately, 

I don't have a timeline for Baselines at this moment but we certainly plan to release them. I saw that you also raised an issue there: https://github.com/openai/baselines/issues/21. There should be some update soon :)

**Comment by miriaford** [comment]
@dementrock is there a timeline for Baseline's next release? Even something vague would be helpful. Like a month from now? Two? Or half a year?

**Comment by BigHopes** [comment]
Can I download rllab on windows? I didn't find that on the internet

**Comment by JACKHAHA363** [comment]
Is there any plan to have a pytorch version?
